docs/user/sbt.rst

```diff
@@ -33,24 +33,35 @@ and now you can write your first application in ``./src/main/scala/HelloWorld.sc
-0.1   ``nativeClang``          ``File``        Path to ``clang`` command
-0.1   ``nativeClangPP``        ``File``        Path to ``clang++`` command
-0.1   ``nativeCompileOptions`` ``Seq[String]`` Extra options passed to clang verbatim during compilation
-0.1   ``nativeLinkingOptions`` ``Seq[String]`` Extra options passed to clang verbatim during linking
-0.1   ``nativeMode``           ``String``      Either ``"debug"`` or ``"release"`` (2)
-0.2   ``nativeGC``             ``String``      Either ``"none"`` or ``"boehm"`` (3)
-===== ======================== =============== =========================================================
+===== ======================== ================ =================================================== =========================================================
+Since Name                     Type             Default Value                                       Description
+===== ======================== ================ =================================================== =========================================================
+0.1   ``nativeClang``          ``Option[File]`` ``None``                                            Path to ``clang`` command
+0.1   ``nativeClangPP``        ``Option[File]`` ``None``                                            Path to ``clang++`` command
+0.1   ``nativeCompileOptions`` ``Seq[String]``  ``Seq("-O0")``                                      Extra options passed to clang verbatim during compilation
+0.1   ``nativeLinkingOptions`` ``Seq[String]``  ``Seq("-I/usr/local/include", "-L/usr/local/lib")`` Extra options passed to clang verbatim during linking
+0.1   ``nativeMode``           ``String``       ``"debug"``                                         Either ``"debug"`` or ``"release"`` (2)
+0.2   ``nativeGC``             ``String``       ``"boehm"``                                         Either ``"none"`` or ``"boehm"`` (3)
+===== ======================== ================ =================================================== =========================================================
+
+Sbt tasks
+----------------------
+
+===== ======================== ============ ====================================================
+Since Name                     Type         Description
+===== ======================== ============ ====================================================
+0.1   ``compile``              ``Analysis`` Compile Scala code to NIR
+0.1   ``run``                  ``Unit``     Compile, link and run the generated binary
+0.1   ``package``              ``File``     Similar to standard package with addition of NIR
+0.1   ``publish``              ``Unit``     Similar to standard publish with addition of NIR (1)
+0.1   ``nativeLink``           ``File``     Link NIR and generate native binary
+0.2   ``nativeFindClang``      ``File``     Path to ``clang`` command
+0.2   ``nativeFindClangPP``    ``File``     Path to ``clang++`` command
```

Comment:
I don't think we need to have any settings at all. We can just make everything a task, this will make sure we don't have any problems with environment discovery in the future. Duplicating every single setting as setting+task is too complicated.